src/python/m5/proxy.py

```diff
@@ -167,9 +167,6 @@ def getindex(obj, index):
     def set_param_desc(self, pdesc):
         self._pdesc = pdesc
 
-    def __hash__(self) -> int:
-        return simobject_hash(str(self))
-
```

Comment:
Can you confirm that this change is needed?

Reply:
your commenting on a commit, not the PR. This change is just reverting the PR back to the code as it currently is on develop

Reply:
Oh, my bad!